test/foo/lib/config/test_configloader.py

```diff
@@ -0,0 +1,55 @@
+from unittest.mock import Mock
+
+import pytest
+
+from foo.lib.config import ConfigLoader, ConfigLoaderException
+
+
+def reset_config_loader_singleton():
+    ConfigLoader._instance = None
+    ConfigLoader._instance_ready = False
+
+
+def test_load(monkeypatch):
+    reset_config_loader_singleton()
+    monkeypatch.setenv('VAR3', 'not bar')
+    cfg_loader = ConfigLoader(defaults={'VAR0': 'default value 0',
+                                        'VAR1': 'default value 1'})
+    config = cfg_loader.load('test/foo/lib/config/config.yaml')
+    assert config._config == {'VAR0': 'fizz',
+                              'VAR1': 'default value 1',
+                              'VAR2': 'foo',
+                              'VAR3': 'not bar'}
+    assert config.source == 'test/foo/lib/config/config.yaml'
+
+
+def test_load_missing():
+    reset_config_loader_singleton()
+    cfg_loader = ConfigLoader()
+    with pytest.raises(ConfigLoaderException):
+        cfg_loader.load('test/foo/lib/config/missing-config.yaml')
+
+
+def test_is_a_singleton():
+    reset_config_loader_singleton()
+    assert ConfigLoader(defaults={'VAR0': 'default value 0'}) is ConfigLoader()
+
+
+def test_load_from_s3(monkeypatch):
```

Comment:
Does this test actually check anything?
the name of this test is misleading because if does not load configuration from real s3 but only mock set of functions and in reality does not check any production-code.

However I would setup production grade s3 and skip differenciation between `prod` and `test` environments and modify this test to upload config file to s3 and later assert that config loaded from s3 matches the config defined in a file created at the beginning of the test.